src/metabase/driver/sql_jdbc/execute.clj

```diff
@@ -690,60 +712,113 @@
   [_ sql remark]
   (str "-- " remark "\n" sql))
 
-(defn execute-reducible-query
+(mu/defn ^:private execute-reducible-query* :- :some
```

Comment:
Didn't touch the logic here, but I split the arity of `execute-reducible-query` that was only used once internally into a separate function to make the code a little more readable.